data/templates/firewall/nftables.j2

```diff
@@ -95,107 +156,128 @@ table ip vyos_filter {
 {% if zone is vyos_defined %}
 {{ zone_tmpl.zone_chains(zone, state_policy is vyos_defined, False) }}
 {% endif %}
-
-{% if state_policy is vyos_defined %}
-    chain VYOS_STATE_POLICY {
-{%     if state_policy.established is vyos_defined %}
-        {{ state_policy.established | nft_state_policy('established') }}
-{%     endif %}
-{%     if state_policy.invalid is vyos_defined %}
-        {{ state_policy.invalid | nft_state_policy('invalid') }}
-{%     endif %}
-{%     if state_policy.related is vyos_defined %}
-        {{ state_policy.related | nft_state_policy('related') }}
-{%     endif %}
-        return
-    }
-{% endif %}
 }
 
 {% if first_install is not vyos_defined %}
 delete table ip6 vyos_filter
 {% endif %}
 table ip6 vyos_filter {
-    chain VYOS_FW6_FORWARD {
-        type filter hook forward priority 0; policy accept;
-{% if state_policy is vyos_defined %}
-        jump VYOS_STATE_POLICY6
-{% endif %}
-{% if interface is vyos_defined %}
-{%     for ifname, ifconf in interface.items() %}
-{%         if ifconf.in is vyos_defined and ifconf.in.ipv6_name is vyos_defined %}
-        iifname {{ ifname }} counter jump NAME6_{{ ifconf.in.ipv6_name }}
-{%         endif %}
-{%         if ifconf.out is vyos_defined and ifconf.out.ipv6_name is vyos_defined %}
-        oifname {{ ifname }} counter jump NAME6_{{ ifconf.out.ipv6_name }}
-{%         endif %}
-{%     endfor %}
-{% endif %}
-        jump VYOS_POST_FW6
+{% if ipv6 is vyos_defined %}
+{%     if ipv6.forward is vyos_defined %}
+{%         set ns = namespace(sets=[]) %}
+{%         for prior, conf in ipv6.forward.items() %}
+{%             set def_action = conf.default_action %}
+    chain VYOS_IPV6_FORWARD_{{ prior }} {
+        type filter hook forward priority {{ prior }}; policy {{ def_action }};
+{%             if conf.rule is vyos_defined %}
+{%                 for rule_id, rule_conf in conf.rule.items() if rule_conf.disable is not vyos_defined %}
+        {{ rule_conf | nft_rule('FWD', prior, rule_id ,'ip6') }}
+{%                     if rule_conf.recent is vyos_defined %}
+{%                         set ns.sets = ns.sets + ['FWD_' + prior + '_' + rule_id] %}
+{%                     endif %}
+{%                 endfor %}
+{%             endif %}
     }
-    chain VYOS_FW6_LOCAL {
-        type filter hook input priority 0; policy accept;
-{% if state_policy is vyos_defined %}
-        jump VYOS_STATE_POLICY6
-{% endif %}
-{% if interface is vyos_defined %}
-{%     for ifname, ifconf in interface.items() %}
-{%         if ifconf.local is vyos_defined and ifconf.local.ipv6_name is vyos_defined %}
-        iifname {{ ifname }} counter jump NAME6_{{ ifconf.local.ipv6_name }}
-{%         endif %}
-{%     endfor %}
-{% endif %}
-        jump VYOS_POST_FW6
+{%         endfor %}
+{%         for set_name in ns.sets %}
+    set RECENT6_{{ set_name }} {
```

Comment:
Do you think the namespace/sets object is better being in global scope and accumulated over the nodes, then iterated only once at the end?